host/src/raiko.rs

```diff
@@ -113,11 +113,9 @@ impl Raiko {
                 );
 
                 // Make sure the blockhash from the node matches the one from the builder
-                check_eq(
-                    &Into::<FixedBytes<32>>::into(header.hash().0),
-                    &input.block_hash_reference,
-                    "block hash unexpected",
-                );
+                if &Into::<FixedBytes<32>>::into(header.hash().0) != &input.block_hash_reference {
```

Comment:
I think it would be more readable to use the `From` trait rather than `Into` like the following:
```rust
if B256::from(header.hash().0) != input.block_hash_reference {
    return Err(RaikoError::Anyhow(anyhow!(
        "block hash unexpected for block {}",
         input.block_number
    )));
}
```